core/new-gui/src/app/workspace/component/operator-panel/operator-panel.component.spec.ts

```diff
@@ -1,16 +1,29 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { OperatorPanelComponent } from './operator-panel.component';
+import { OperatorMetadataService } from '../../service/operator-metadata/operator-metadata.service';
+import { StubOperatorMetadataService } from '../../service/operator-metadata/stub-operator-metadata.service';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { GroupInfo, OperatorSchema } from '../../types/operator-schema';
+
+import { MOCK_OPERATOR_METADATA, MOCK_OPERATOR_GROUPS, MOCK_OPERATOR_SCHEMA_LIST } from '../../service/operator-metadata/mock-operator-metadata.data';
+
+import * as c from './operator-panel.component';
+import '../../../common/rxjs-operators';
 
 describe('OperatorPanelComponent', () => {
   let component: OperatorPanelComponent;
   let fixture: ComponentFixture<OperatorPanelComponent>;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ OperatorPanelComponent ]
+      declarations: [OperatorPanelComponent],
+      providers: [
+        { provide: OperatorMetadataService, useClass: StubOperatorMetadataService }
```

Comment:
F2FQ: so which class are we using operator or stuboperator?

Reply:
OperatorMetadataService is the provided class we try to mock, StubOperatorMetadataService is the actual class used to mock that class.